x/ibc/23-commitment/types/merkle.go

```diff
@@ -118,34 +128,51 @@ func (mp MerklePath) Pretty() string {
 
 // IsEmpty returns true if the path is empty
 func (mp MerklePath) IsEmpty() bool {
-	return len(mp.KeyPath) == 0
+	return len(mp.KeyPaths) == 0
 }
 
 // ApplyPrefix constructs a new commitment path from the arguments. It interprets
 // the path argument in the context of the prefix argument.
 //
+// ApplyPrefix will prepend a new Merkle KeyPath to the existing KeyPaths. This
+// allows a single path to be represent a chained merkle proof
+//
 // CONTRACT: provided path string MUST be a well formated path. See ICS24 for
 // reference.
-func ApplyPrefix(prefix exported.Prefix, path string) (MerklePath, error) {
-	err := host.DefaultPathValidator(path)
+func ApplyPrefix(prefix exported.Prefix, path exported.Path) (exported.Path, error) {
```

Comment:
ApplyPrefix has changed to use the prefix to prepend a merkle path for the next tree in a chained proof. This corresponds with the current usage. Since the prefix `ibc` would be the path for the next tree in the chained multistore proof proving that the iavl roothash for `ibc` substore is included in multistore hash